src/ui-test/extension.test.ts

```diff
@@ -0,0 +1,416 @@
+	maven.stdoutLineReader.on('line', console.log);
+
+	return maven.wait();
+}
+
+function executeProject(runtime: Runtime, camelVersion: string): Maven {
+	const maven = new Maven({
+		args: [mavenGoals[runtime]],
+		properties: {
+			'camel.version': camelVersion
+		},
+		cwd: WORKSPACE_PATH
+	});
+
+	maven.spawn();
+	return maven;
+}
+
+async function analyzeProject(maven: Maven): Promise<RuntimeOutput> {
+	const analyzer = new LogAnalyzer(maven.stdoutLineReader);
+
+	analyzer.whenMatchesThenCaptureData(/.*Total (?<totalRoutes>\d+) routes, of which (?<startedRoutes>\d+) are started/);
+	analyzer.whenMatchesThenCaptureData(
+		/.*Apache Camel (?<camelVersion>\d+\.\d+\.\d+(|\.[a-zA-Z0-9-_]+)) \(CamelContext: .+\) started in.*/
+	);
+	analyzer.startOrderedParsing();
+
+	const analyzerResult = await analyzer.wait() as RuntimeOutput;
+	return analyzerResult;
+}
+
+function getExpectedNumberOfRoutes(args: TestArguments): number {
+	switch (args.type) {
+		case 'file':
+			return 10;
+		case 'url':
+			return 2;
+		default:
+			expect.fail('Unsupported option');
+			return -1;
+	}
+}
+
+function getCamelContextPath(args: TestArguments): string {
+	switch (args.framework) {
+		case 'spring':
+			return 'src/main/resources/META-INF/spring/camel-context.xml';
+		case 'blueprint':
+			return 'src/main/resources/OSGI-INF/blueprint/blueprint.xml';
+		default:
+			expect.fail('Unsupported option');
+			return null;
+	}
+}
+
+function getSourceRootOfGeneratedFiles(args: TestArguments): string {
+	switch (args.type) {
+		case 'file':
+			return '/src/main/java/org/jboss/fuse/wsdl2rest/test/doclit';
+		case 'url':
+			return 'src/main/java/org/helloworld/test/rpclit';
+		default:
+			expect.fail('Unsupported option');
+			return null;
+	}
+}
+
+function getExpectedFileList(args: TestArguments): string[] {
+	let files = [
+		'wsdl2rest.readme.md',
+		'config/logging.properties',
+		getCamelContextPath(args)
+	];
+
+	const sourceRoot: string = getSourceRootOfGeneratedFiles(args);
+	switch (args.type) {
+		case 'file':
+			files.push(
+				`${sourceRoot}/AddAddress.java`,
+				`${sourceRoot}/AddAddressResponse.java`,
+				`${sourceRoot}/Address.java`,
+				`${sourceRoot}/AddressService.java`,
+				`${sourceRoot}/DelAddress.java`,
+				`${sourceRoot}/DelAddressResponse.java`,
+				`${sourceRoot}/GetAddress.java`,
+				`${sourceRoot}/GetAddressResponse.java`,
+				`${sourceRoot}/Item.java`,
+				`${sourceRoot}/ListAddresses.java`,
+				`${sourceRoot}/ListAddressesResponse.java`,
+				`${sourceRoot}/ObjectFactory.java`,
+				`${sourceRoot}/package-info.java`,
+				`${sourceRoot}/UpdAddress.java`,
+				`${sourceRoot}/UpdAddressResponse.java`,
+			);
+			break;
+		case 'url':
+			files.push(
+				`${sourceRoot}/HelloPortType.java`,
+				`${sourceRoot}/HelloService.java`,
+			);
+			break;
+		default:
+			expect.fail('Unsupported option');
+			return null;
+	}
+	return files;
+}
+
+async function getInput(driver: WebDriver): Promise<Input> {
+	// avoid sleep conditions
```

Comment:
I think this comment can be removed, I guess the sleep conditions has been avoided

Reply:
+1